conpot/core/protocol_wrapper.py

```diff
@@ -0,0 +1,29 @@
+from conpot.core import get_interface
+from datetime import datetime
+
+core_interface = get_interface()
+
+
+def conpot_protocol(cls):
+    class Wrapper(object):
+        def __init__(self, *args, **kwargs):
+            self.wrapped = cls(*args, **kwargs)
+            self.cls = cls
+            if self.cls.__name__ not in 'Proxy':
+                core_interface.protocols[self.cls] = self.wrapped
+
+        def __getattr__(self, name):
```

Comment:
Sorcery using  magic methods. 
Basically implementing __getattr__ overrides Python's default mechanism for member access. So basically every attribute of a class that is decorated by `@conpot_protocol` decorator can be accessed here.  This can be very powerful in case we want to emulate a system-wide phenomenon. Like for example we want to emulate a system restart (kamstrup management protocol ;-) we can set a counter here and freeze access to all protocols. 

Some other uses include timing the last attack. This can be done by tracking the handle method for every protocol. Again can be easily done from here, ***without even touching the protocol implementation :-)***.

Reply:
that sounds really nice to me -
@glaslos would you mind to think this through for a second? :)

Reply:
@creolis, @glaslos - For the downsides of this approach, please have a look at my comment below